hack/loadtest/util/placement.go

```diff
@@ -79,148 +93,227 @@ func MeasureOnePlacement(ctx context.Context, hubClient client.Client, deadline,
-					Group:   "",
-					Version: "v1",
-					Kind:    "Namespace",
-					LabelSelector: &metav1.LabelSelector{
-						MatchLabels: map[string]string{labelKey: nsName},
-					},
-				},
-				{
-					Group:   apiextensionsv1.GroupName,
-					Version: "v1",
-					Kind:    "CustomResourceDefinition",
-					Name:    "clonesets.apps.kruise.io",
-				},
-			},
-		},
+	crp := &v1beta1.ClusterResourcePlacement{}
+	if err := createCRP(crp, crpFile, crpName, nsName); err != nil {
+		klog.ErrorS(err, "failed to create crp", "namespace", nsName, "crp", crpName)
+		return err
 	}
+
 	defer hubClient.Delete(context.Background(), crp) //nolint
 	if err := hubClient.Create(ctx, crp); err != nil {
 		klog.ErrorS(err, "failed to apply crp", "namespace", nsName, "crp", crpName)
 		return err
 	}
+	crpCount.Inc()
 
 	klog.Infof("verify that the cluster resource placement `%s` is applied", crpName)
-	collectApplyMetrics(ctx, hubClient, deadline, interval, crpName, currency, fleetSize)
+	fleetSize, clusterNames = collectApplyMetrics(ctx, hubClient, deadline, interval, crpName, currency, fleetSize, clusterNames)
 
-	klog.Infof("remove the namespaced resources referred by the placement `%s`", crpName)
+	klog.Infof("remove the namespaced resources applied by the placement `%s`", crpName)
+	deletionStartTime := time.Now()
 	if err := deleteTestManifests(ctx, hubClient, nsName); err != nil {
 		klog.ErrorS(err, "failed to delete test manifests", "namespace", nsName, "crp", crpName)
 		return err
 	}
 	collectDeleteMetrics(ctx, hubClient, deadline, interval, crpName, clusterNames, currency, fleetSize)
 
+	// wait for the status of the CRP and make sure all conditions are all true
+	klog.Infof("verify cluster resource placement `%s` is updated", crpName)
+	waitForCrpToComplete(ctx, hubClient, deadline, interval, deletionStartTime, crpName, currency, fleetSize)
 	return hubClient.Delete(ctx, crp)
 }
 
 // collect the crp apply metrics
-func collectApplyMetrics(ctx context.Context, hubClient client.Client, deadline, pollInterval time.Duration, crpName string, currency string, fleetSize string) {
+func collectApplyMetrics(ctx context.Context, hubClient client.Client, deadline, pollInterval time.Duration, crpName string, currency string, fleetSize string, clusterNames ClusterNames) (string, ClusterNames) {
 	startTime := time.Now()
 	applyDeadline := startTime.Add(deadline)
-	var crp v1alpha1.ClusterResourcePlacement
+	var crp v1beta1.ClusterResourcePlacement
 	var err error
-	for ; ctx.Err() == nil; time.Sleep(pollInterval) {
-		if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
-			klog.ErrorS(err, "failed to get crp", "crp", crpName)
-			if time.Now().After(applyDeadline) {
-				// timeout
-				klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
-				LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
-				applyTimeoutCount.Inc()
-				break
+	ticker := time.NewTicker(pollInterval)
```

Comment:
Hi Britania! Just out of curiosity, is there any reason why this one is added without a timer?

Reply:
No I guess I forgot to add it. Did not realize.